src/sentry/models/group.py

```diff
@@ -369,6 +369,41 @@ def by_qualified_short_id_bulk(
             ).filter(short_id_lookup, project__organization=organization_id)
         )
         group_lookup: set[int] = {group.short_id for group in groups}
+
+        # If any requested short_ids are missing after the exact slug match,
+        # fallback to a case-insensitive slug lookup to handle legacy/mixed-case slugs.
+        missing_short_ids = [sid.short_id for sid in short_ids if sid.short_id not in group_lookup]
+        if missing_short_ids:
+            # Build a lookup only for the missing short_ids per slug
+            missing_by_slug = defaultdict(list)
+            for sid in short_ids:
+                if sid.short_id in missing_short_ids:
+                    missing_by_slug[sid.project_slug].append(sid.short_id)
```

Comment:
i think you hold all the missing short ids already in missing_short_ids?
```suggestion
            for missing_short_id in missing_short_ids
                    missing_by_slug[short_id.project_slug].append(short_id.short_id)
```